packages/base-controller/src/BaseControllerV2.ts

```diff
@@ -248,15 +248,28 @@ function deriveStateFromMetadata<S extends Record<string, Json>>(
   metadataProperty: 'anonymous' | 'persist',
 ): Record<string, Json> {
   return Object.keys(state).reduce((persistedState, key) => {
```

Comment:
Note for later: this variable name is misleading, as this is used for the anonymized state derivation function as well. I'll follow-up to rename this to "derivedState".